src/_guides/language/coming-from/js-to-dart.md

```diff
@@ -2632,88 +2816,69 @@ indicate they are initialized at a later point,
 while remaining non-nullable. This is useful
 for cases where a variable is never observed
 before being initialized, allowing it to be initialized later.
-This has several advantages over just labeling the field as nullable:
+This differs from labeling the field as nullable:
 
 * (Non-nullable) late fields cannot have null assigned at a later point.
 
 * (Non-nullable) late fields throw a runtime error when
-  accessed before they are initialized.
+  accessed before they initialize. This should be avoided.
 
 {:.include-lang}
 ```dart
-// Using null safety:
-class Coffee {
-  late String _temperature;
-
-  void heat() { _temperature = 'hot'; }
-  void chill() { _temperature = 'iced'; }
-
-  String serve() => _temperature + ' coffee';
+class PetOwner {
+  final String name;
+  late final Pet _pet;
+  PetOwner(this.name, String petName) {
+    // Cyclic object graph, cannot set _pet before owner exists.
+    _pet = Pet(petName, this);
+  }
+  Pet get pet => _pet;
+}
+class Pet {
+  final String name;
+  final PetOwner owner;
+  Pet(this.name, this.owner);
 }
 ```
 
-In this case, `_temperature` is only initialized
-after calling `heat()` or `chill()`.
-If `serve()` is called before either are called,
-a runtime exception occurs.
-The `_temperature` field can never be assigned `null`.
+Use `late` for local variables only if unclear code results
+in the compiler being unable determine if the code initialized the variable.
 
-You can use the `late` keyword to make initialization _lazy_,
-when combined with an initializer:
-
-{:.include-lang}
 ```dart
-class Weather {
-  late int _temperature = _readThermometer();
+doSomething(int n, bool capture) {
+  late List<Foo> captures;
+  if (capture) captures = [];
+  for (var i = 0; i < n; i++) {
+    var foo = something(i);
+    if (capture) captures.add(foo);
+  }
 }
-```
 
-In this example, `_readThermometer()` only runs
-when the field is first accessed,
-rather than on initialization. 
+In the preceding example, the compiler does not know to assign
+`captures` if `capture` is true. In this case `capture` is true,
```

Comment:
Drop "In this case `capture` is true,", we don't know that. The logic is sound whether it's true or not.